src/main/core/Subscriber/SecurityEventSubscriber.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Claroline\CoreBundle\Subscriber;
+
+use Claroline\CoreBundle\Entity\Log\LogSecurity;
+use Claroline\CoreBundle\Event\CatalogEvents\SecurityEvents;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class SecurityEventSubscriber implements EventSubscriberInterface
+{
+    private $em;
+    private $client;
+    private $security;
+
+    public function __construct(
+        EntityManagerInterface $em,
+        HttpClientInterface $client,
+        Security $security
+    ) {
+        $this->em = $em;
+        $this->client = $client;
+        $this->security = $security;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            SecurityEvents::USER_LOGIN => 'logEvent',
+            SecurityEvents::USER_LOGOUT => 'logEvent',
+            SecurityEvents::USER_DISABLE => 'logEvent',
+            SecurityEvents::USER_ENABLE => 'logEvent',
+            SecurityEvents::NEW_PASSWORD => 'logEvent',
+            SecurityEvents::FORGOT_PASSWORD => 'logEvent',
+            SecurityEvents::ADD_ROLE => 'logEvent',
+            SecurityEvents::REMOVE_ROLE => 'logEvent',
+            SecurityEvents::VIEW_AS => 'logEvent',
+        ];
+    }
+
+    public function logEvent($event): void
+    {
+        $request = Request::createFromGlobals();
```

Comment:
This is unnecessarily expensive. We should retrieve the current request by calling `getCurrentRequest()` on the `RequestStackInterface` service

Reply:
Change done, thx :)